src/credentials/activation.ts

```diff
@@ -31,15 +32,21 @@ export async function loginWithMostRecentCredentials(
     toolkitSettings: SettingsConfiguration,
     loginManager: LoginManager
 ): Promise<void> {
+    const profiles = await CredentialsProviderManager.getInstance().getProfiles()
+    const profileNames = Object.keys(profiles)
     const previousCredentialsId = toolkitSettings.readSetting<string>(profileSettingKey, '')
-    if (previousCredentialsId) {
+    if (profiles && profileNames.length === 1) {
+        // Auto-connect if there is exactly one profile, named "default".
+        await loginManager.login(profiles[profileNames[0]])
+        // Toast.
+        vscode.window.showInformationMessage(`Connected to "${profileNames[0]}"`)
+    } else if (previousCredentialsId) {
```

Comment:
I would have this as the first checked case, with the new case as the `else if`. We don't need to inform users who are coming back that they're reconnected under the same credentials.

Reply:
I agree with this. Also, this would maintain the existing flow where "a user is logged out if the toolkit tries to use last sessions credentials but they no longer exist"